pricegraph/src/api/price_source.rs

```diff
@@ -0,0 +1,89 @@
+//! This module implements price source methods for the `Pricegraph` API so that
+//! it can be used for OWL price estimates to the solver.
+
+use crate::encoding::{TokenId, TokenPair};
+use crate::{Pricegraph, FEE_TOKEN};
+
+const OWL_BASE_UNIT: f64 = 1_000_000_000_000_000_000.0;
+
+impl Pricegraph {
+    /// Estimates the fee token price in WEI for the specified token. Returns
+    /// `None` if there token is not connected to the fee token (that is, there
+    /// is no transitive order buying the fee token for the specified token).
+    ///
+    /// The fee token is defined as the token with ID 0.
+    pub fn estimate_token_price(&self, token: TokenId) -> Option<f64> {
+        if token == FEE_TOKEN {
+            return Some(OWL_BASE_UNIT);
+        }
+
+        // Estimate price of selling 1 unit of the reference token for each
+        // token. We sell rather than buy the reference token because volume is
```

Comment:
"for each token" sounds strange because there is only one token. (In the pricegraph based price source this made sense.)

Reply:
Hehe just copy pasted comment over, nice catch!